packages/mobile/src/store/offline-downloads/sagas.ts

```diff
@@ -86,8 +108,71 @@ function* watchClearOfflineDownloads() {
   yield* takeLatest(clearOfflineDownloads, clearOffineDownloadsAsync)
 }
 
+export function* startSync() {
+  try {
+    yield* take(doneLoadingFromDisk)
+    yield* waitForRead()
+    yield* waitForBackendSetup()
+    const collections = yield* select(getCollections)
+    // Don't use getAccountSelections as it filters out collections not in cache
+    const accountCollections: { [id: number]: AccountCollection } =
+      yield* select((state) => state.account.collections)
+    const accountCollectionIds = Object.values(accountCollections).map(
+      (collection) => collection.id
+    )
+    const offlineCollectionsState = yield* select(getOfflineCollections)
+    const existingOfflineCollections: Collection[] = Object.entries(
+      offlineCollectionsState
+    )
+      .filter(
+        ([id, isDownloaded]) => isDownloaded && id !== DOWNLOAD_REASON_FAVORITES
+      )
+      .map(([id, isDownloaded]) => collections[id] ?? null)
+      .filter((collection) => !!collection)
+
+    const isFavoritesDownloadEnabled =
+      offlineCollectionsState[DOWNLOAD_REASON_FAVORITES]
+    if (isFavoritesDownloadEnabled) {
+      // Individual tracks
+      yield* call(syncFavoritedTracks)
+
+      // Favorited collections
+      for (const collectionId of accountCollectionIds) {
+        yield* put(fetchCollection(collectionId))
```

Comment:
so just want to make sure the impact here isn't too great on startup if we are making alot of requests?

Reply:
Yeah I was thinking about how to mitigate that. Might be okay to add a short delay here between requests so they never stack up?

Reply:
yeah i think potentially we just see what the perf looks like? so in this case, can we just wait until the user navigates to the collections to fire this off?

Reply:
i guess this is only a one-time thing though? or are we continuously looking for updates?

Reply:
Just once per session. All we need is to generally keep your offline content up to date. If we miss something, we'll do it next session. Thought about adding some sort of loop in case someone leaves the app open for months, but I'm not so worried about that edge case

Reply:
Yeah sounds good. Any tips for measuring perf impact with flipper? I can try an account with several collections and see if it scores lower on the time to response test.

Impact of the sync process should be:
- 1 request per collection the user has owned/favorited
- delete operations for removed tracks
- any missing tracks enqueued for download
- overhead of calculating what's missing in the above

it's linear wrt the number and length of collections, and the largest component (downloads) is deferred to the queue

Reply:
https://linear.app/audius/issue/C-1855/ensure-sync-doesnt-cause-performance-hit-on-app-load

Making a follow-up ticket so I can get this merged and tested sooner.

Reply:
yeah with flipper, one potential easy win is to try the react perf plugin, the one that gives lighthouse like scores. we can try that together when this is played